src/object/cli_shard.c

```diff
@@ -1955,7 +1955,7 @@ obj_shard_query_key_cb(tse_task_t *task, void *data)
 			cb_args->dkey->iov_len = okqo->okqo_dkey.iov_len;
 			changed = true;
 		} else if (flags & DAOS_GET_MAX) {
-			if (*val > *cur) {
+			if (*val >= *cur) {
```

Comment:
"if (*val > *cur)" means dkey changed, and "==" is same dkey right?
why change to ">=" above?  and if you change to ">=" for "flags & DAOS_GET_MAX" case, why did not change it for below "flags & DAOS_GET_MIN" case?

or maybe need not change here, but change below L2002 to be "if (changed || check)"?

Reply:
oh, we want to check all shards to figure out size, if it only check >, then only 1 shard might be check, for example  2 + 1(cell size = 1M),  the whole extent is [0-3.5M],  then shard 0 [0-2M], shard 1 [0-1.5M], and they will shard the same dkey here, so if shard 0 is checked first, then shard 1 will not be checked,   so the size will become 0-3M, not 3.5M.

Reply:
can it be satisfied by remove "if (changed)" below L2002 (or change it to be "if (changed || check)"?

the change as you made, seems will cause some difference for "if (check && flags & DAOS_GET_AKEY) " section's handling?

Reply:
/** if first cb, or dkey changed, set akey */
		if (first || changed)
			*cur = *val;
I thought this will only make *cur to be assigned again with the same value? or I miss sth?